iceoryx2-ffi/ffi/src/api/node_id.rs

```diff
@@ -0,0 +1,220 @@
+///
+/// # Safety
+///
+/// * `node_id_struct_ptr` - Must be either a NULL pointer or a pointer to a valid [`iox2_node_id_t`].
+///   If it is a NULL pointer, the storage will be allocated on the heap.
+/// * `node_id_ptr` - Must be a valid [`iox2_node_id_ptr`]
+/// * `node_id_handle_ptr` - Must point to a valid [`iox2_node_id_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_node_id_clone_from_ptr(
+    node_id_struct_ptr: *mut iox2_node_id_t,
+    node_id_ptr: iox2_node_id_ptr,
+    node_id_handle_ptr: *mut iox2_node_id_h,
+) {
+    debug_assert!(!node_id_handle_ptr.is_null());
+    debug_assert!(!node_id_ptr.is_null());
+
+    *node_id_handle_ptr = std::ptr::null_mut();
+
+    let mut node_id_struct_ptr = node_id_struct_ptr;
+    fn no_op(_: *mut iox2_node_id_t) {}
+    let mut deleter: fn(*mut iox2_node_id_t) = no_op;
+    if node_id_struct_ptr.is_null() {
+        node_id_struct_ptr = iox2_node_id_t::alloc();
+        deleter = iox2_node_id_t::dealloc;
+    }
+    debug_assert!(!node_id_struct_ptr.is_null());
+
+    unsafe {
+        (*node_id_struct_ptr).deleter = deleter;
+    }
+
+    (*node_id_struct_ptr).value.init(*node_id_ptr);
+    *node_id_handle_ptr = (*node_id_struct_ptr).as_handle();
+}
+
+/// Creates a new [`iox2_node_id_h`] by cloning a [`iox2_node_id_h_ref`].
+///
+/// # Safety
+///
+/// * `node_id_struct_ptr` - Must be either a NULL pointer or a pointer to a valid [`iox2_node_id_t`].
+///   If it is a NULL pointer, the storage will be allocated on the heap.
+/// * `node_id_handle` - Must be a valid [`iox2_node_id_h_ref`]
+/// * `node_id_handle_ptr` - Must point to a valid [`iox2_node_id_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_node_id_clone_from_handle(
+    node_id_struct_ptr: *mut iox2_node_id_t,
+    node_id_handle: iox2_node_id_h_ref,
+    node_id_handle_ptr: *mut iox2_node_id_h,
+) {
+    node_id_handle.assert_non_null();
+    debug_assert!(!node_id_handle_ptr.is_null());
+
+    let node_id = &mut *node_id_handle.as_type();
+    let node_id_ptr = node_id.value.as_ref();
+
+    iox2_node_id_clone_from_ptr(node_id_struct_ptr, node_id_ptr, node_id_handle_ptr);
+}
+
+/// Returns the high bits of the underlying value of the [`iox2_node_id_h`].
+///
+/// # Safety
+///
+/// * `node_id_handle` - Must be a valid [`iox2_node_id_h_ref`]
+#[no_mangle]
+pub unsafe extern "C" fn iox2_node_id_value_high(node_id_handle: iox2_node_id_h_ref) -> u64 {
+    node_id_handle.assert_non_null();
+
+    let node_id = &mut *node_id_handle.as_type();
+    (node_id.value.as_ref().value() >> 64) as u64
+}
+
+/// Returns the low bits of the underlying value of the [`iox2_node_id_h`].
+///
+/// # Safety
+///
+/// * `node_id_handle` - Must be a valid [`iox2_node_id_h_ref`]
+#[no_mangle]
+pub unsafe extern "C" fn iox2_node_id_value_low(node_id_handle: iox2_node_id_h_ref) -> u64 {
+    node_id_handle.assert_non_null();
+
+    let node_id = &mut *node_id_handle.as_type();
+    ((node_id.value.as_ref().value() << 64) >> 64) as u64
```

Comment:
Shouldn't this also work? AFAIK, `as` is not saturating
```suggestion
    node_id.value.as_ref().value() as u64
```